src/main/java/io/appform/dropwizard/actors/base/UnmanagedPublisher.java

```diff
@@ -106,6 +113,79 @@ public final long pendingSidelineMessagesCount() {
         return Long.MAX_VALUE;
     }
 
+    /**
+     * Note: Timeout is in MilliSeconds, Function take a list of message as input and return not ack msg as output
+     * @param messages : Messages to be published
+     * @param properties
+     * @param timeout : in MS timeout for waiting on countDownLatch
+     * @param unit : timeout unit
+     * @return : List of message nacked
+     * @throws Exception
+     */
+    public final List<Message> publishWithConfirmListener(List<Message> messages, AMQP.BasicProperties properties,
+            long timeout, @NotNull TimeUnit unit) throws Exception {
+        publishChannel.confirmSelect();
+        ConcurrentNavigableMap<Long, Message> outstandingConfirms = new ConcurrentSkipListMap<>();
+        final CountDownLatch publishAckLatch = new CountDownLatch(messages.size());
+        publishChannel.addConfirmListener((sequenceNumber, multiple) -> {
+            if (multiple) {
+                ConcurrentNavigableMap<Long, Message> confirmed = outstandingConfirms.headMap(
+                        sequenceNumber, true
+                );
+                for(int i =0;i<confirmed.size();i++)
+                    publishAckLatch.countDown();
+                confirmed.clear();
+            } else {
+                publishAckLatch.countDown();
+                outstandingConfirms.remove(sequenceNumber);
+            }
+            log.info(String.format("Message with delivery tag %s acknoledged", sequenceNumber));
+        }, (sequenceNumber, multiple) -> {
+            {
+                if(multiple == true)
+                {
+                    ConcurrentNavigableMap<Long, Message> confirmed = outstandingConfirms.headMap(
+                            sequenceNumber, true
+                    );
+                    for(int i =0;i<confirmed.size();i++)
+                        publishAckLatch.countDown();
+                }
+                else
+                    publishAckLatch.countDown();
```

Comment:
if the messages that are published were 1-10 seq number.
If this nacked lambda is called with (4, true), (6, true) won't it countDown to 10?

Reply:
multiple true mean, till that sequence number all the msg are ack/nack.  so lets say List<sequencenumber> = {1,2,3,4,5}


if callback return (4,true) then all the msg {1,2,3,4} are ack/nack

so (4,true) delete ack {1,2,3,4} and (6,true) ack {5,6}

Reply:
We are not calling confirmed.clear() in nack case  so confirmed.size() will return 4 and 6 right?
In case of ack case, it will give 4 and 2

Reply:
fixed